modeling/03_Per_Sat_Model.Rmd

```diff
@@ -0,0 +1,1047 @@
+---
+title: "eePlumB per-Satellite Model"
+author: "ROSSyndicate"
+date: "2023-07-20"
+output: html_document
+---
+
+```{r setup, echo = F}
+libs = c('reticulate', 'tidyverse')
+
+package_loader <- function(x) {
+    if (x %in% installed.packages()) {
+      library(x, character.only = TRUE)
+    } else {
+      install.packages(x)
+      library(x, character.only = TRUE)
+    }
+}
+
+lapply(libs, package_loader)
+```
+
+## Activate conda environment
+
+Check for virtual environment and activate, otherwise, set up virtual
+environment.
+
+```{r conda env}
+if (!dir.exists("env")) {
+  source("pySetup.R")
+} else {
+  use_condaenv(file.path(getwd(), "env"))
+}
+```
+
+And then import the needed modules
+
+```{python}
+import ee
+import os
+import time
+import pandas as pd
+import fiona
+```
+
+## GEE Setup
+
+In order to access the GEE API, we need to authenticate our identity. To do
+this, you must type `earthengine authenticate` in the terminal window. When
+your browser states 'You are now authenticated with the gcloud CLI', the
+authentication is complete. This authentication is valid for 7 days.
+
+Now, we need to initialize our GEE session.
+
+```{python}
+ee.Initialize(project = 'ee-ross-superior')
+```
+
+# Import assets
+
+## Load the labels file into the environment
+
+Because of how the later function is written, it works best to read these into R. I couldn't tell you why other than my function doesn't work with pandas dataframes.
+
+```{r}
+labels_file <- list.files('data/labels/', full.names = T)
+labels <- read_csv(labels_file)
+
+# subset for sen/ls
+sen_labels = labels %>% 
+  filter(mission == 'SEN2')
+ls_labels = labels %>% 
+  filter(mission != 'SEN2')
+
+# and for 5/7/8/9
```

Comment:
Did we drop `LS4` ?

Reply:
yup! There are only five LS4 dates, if you think it's really important, I'd likely apply the LS5 GTB model to LS4 - even if we label those 4 images, we're likely to not have a robust enough model for them.